src/assets/alignment-D3.js

```diff
@@ -0,0 +1,186 @@
+    getTickPosition
+  );
+
+  return _isShapeValueWithinTickValues(
+    shape,
+    enclosingTicks,
+    getShapeValue,
+    getTickValue
+  );
+}
+
+// Gets the nearest tick to a given position.
+// The way it does this is not obvious. First it filters all the ticks to only
+// get the ticks before or after the given position. The filtering is based
+// on the "filterCompare" function parameter. Then it performs a reduce on the
+// filtered ticks to find which one is closest to the given postion. The reduce
+// function uses the "compare" parameter which is a function to compare ticks.
+// See the "_getSurroundingTicks" function for an example of how this is used.
+function getNearestTick(
+  allTicks,
+  filterCompare,
+  dimension,
+  position,
+  getTickPosition,
+  compare
+) {
+
+  // Finds the tick that is closest to the given position, based on the
+  // compare function.
+  const reduceFunction = function(result, tick) {
+    const position = getTickPosition(tick)[dimension];
+    if (result && compare(getTickPosition(result)[dimension], position)) {
+      return result;
+    }
+    return tick;
+  };
+
+  // First filter the ticks to get only the ticks that are before or after
+  // the given position.
+  let ticks = allTicks.filter(tick => {
+    let tickPosition = getTickPosition(tick)[dimension];
+    return filterCompare(tickPosition, position[dimension]);
+  });
+
+  // Finally, run the reduce operation to get the closest tick.
+  let closestTick = ticks.reduce(reduceFunction, null);
+
+  return closestTick;
+}
+
+// TODO: Only exported so we can test it.
+// Given a list of ticks it will find the ticks closest to the given position.
+// This also works when there is no beforeTick or afterTick. I.e. sometimes some
+// of the small values appear before the first tick, or the largest values
+// appear after the last tick. In those cases it will return null for the
+// tick in question.
+export function _getSurroundingTicks(
+  ticksList,
+  dimension,
+  position,
+  getTickPosition,
+) {
+
+  let ticks;
+  let afterTick;
+  let beforeTick;
+
+  if (!ticksList) {
+    throw new Error('The list of ticks must not be empty.');
+  }
+
+  ticks = [].slice.call(ticksList);
+
+  // The filter function finds all ticks less than or equal to the position.
+  // The compare function then finds the largest of the filtered tickss.
+  beforeTick = getNearestTick(
+    ticks,
+    (tickPosition, position) => tickPosition <= position,
+    dimension,
+    position,
+    getTickPosition,
+    (prevPosition, position) => prevPosition > position
+  );
+
+  // The filter function finds all ticks greater than the position.
+  // The compare function then finds the smallest of the filtered tickss.
+  afterTick = getNearestTick(
+    ticks,
+    (tickPosition, position) => tickPosition > position,
+    dimension,
+    position,
+    getTickPosition,
+    (prevPosition, position) => prevPosition < position
+  );
+
+  return [ beforeTick, afterTick ];
```

Comment:
The way accounting for axis sort order was handled before was to assign the lower value to beforeTick and vice versa. i.e. `beforeTick = Math.min(beforeTick, afterTick)`

Reply:
@tbushman I'm pretty sure the way I calculate the beforeTick and afterTick already account for this. It doesn't matter what order the ticks are on the axis. For the beforeTick for example, it just looks for all the ticks with a value that lower than the shape value, and then it finds the tick closest to the shape value. I'll double check it when I get a chance.